### ISSUE TITLE
define list of RDF serialization formats

### ISSUE THREAD

**Issue by VladimirAlexiev** [w3c/rdf-concepts#83]
Try as I might, I cannot find a list of RDF serialization formats.
- SPARQL Protocol: https://www.w3.org/TR/sparql12-protocol/#query-success points to specs for the 4 tabular formats suitable for SELECT/ASK
  - But for CONSTRUCT and DESCRIBE it gives only RDF XML as an example, then refers to RDF Concepts for more
- RDF Concepts: https://www.w3.org/TR/rdf12-concepts/#section-dataset-conneg doesn't have a list of formats
- RDF 1.1 Primer: https://www.w3.org/TR/rdf11-primer/#section-graph-syntax has a great list of examples and links for 7 formats, but is not updated since 2014

It would be good to have an official list of formats, with links to their respective specs.
I don't know whether this belongs in RDF Concepts, but is there any better place?

https://graphdb.ontotext.com/documentation/10.5/rdf-formats.html lists the formats and MIME types supported by rdf4j. 
Here is a subset excluding RDF-Star formats and alternative MIME types:

| Format    | Content-Type          | note                                                             |
|-----------|-----------------------|------------------------------------------------------------------|
| RDF XML   | application/rdf+xml   |                                                                  |
| Turtle    | text/turtle           |                                                                  |
| N-Triples | application/n-triples |                                                                  |
| N-Quads   | application/n-quads   |                                                                  |
| JSON-LD   | application/ld+json   |                                                                  |
| RDF JSON  | application/rdf+json  |                                                                  |
| N3        | text/n3               | Extension of Turtle, but typical repositories treat it as Turtle |
| TriX      | application/trix      |                                                                  |
| TriG      | application/trig      |                                                                  |

A major omission is RDFa and Microdata.

**Comment by csarven** [comment]
https://www.w3.org/TR/rdf12-concepts/#dfn-concrete-rdf-syntax normatively lists some example concrete RDF syntaxes.

On a related note, there is also https://www.w3.org/ns/formats/ but that is not specifically about concrete RDF syntaxes.

That said, there could be a separate Living Document (or similar) that lists all concrete RDF syntaxes which can then be referenced from RDF-CONCEPTS (and elsewhere).

AFAIK, Microdata is not a classified as a "concrete RDF syntax" - although it could be transformed to a concrete RDF syntax, which is a separate step/matter, just as anything else that could potentially be transformed to RDF.

RDFa (in markup languages) is a concrete RDF syntax.

**Comment by csarven** [comment]
.. the server can't divine any concrete RDF syntax out of thin air (irrespective to host language being involved or not). There is no need to introduce dimensions whatsovever focusing on reading or writing because that's orthogonal to what a concrete RDF syntax entails. And again, in the same way we don't introduce things like whether a concrete RDF syntax is humand- *and* machin-readable or not, and whether actual consumption of certain concrete RDF syntaxes actually require *additional* processing or requiring JavaScript or so forth to be actually usable by a human.

**Comment by pfps** [comment]
Shouldn't this list go in the primer, not in concepts?

**Comment by VladimirAlexiev** [comment]
@csarven 
> The server cannot divine some HTML out of thin air?

I mean this: for all formats not related to HTML, given an RDF dataset, it is clear how to serialize to that format.
That's not the case about RDFa because it requires some HTML elements and text: how would the server divine them?

**Comment by pfps** [comment]
My view is that this is not something for the RDF Concepts document.  In fact, https://www.w3.org/TR/rdf11-primer/ has this list already.

**Comment by hartig** [comment]
I agree with @pfps. Such a list should be in the primer and not in RDF Concepts.

**Comment by VladimirAlexiev** [comment]
@pfps if you read my original comment, I cited the Primer. It lacks:
- summary table
- MIME types (but maybe I haven't looked close enough)
- TriX
- RDF JSON

Could you please reopen this issue to add the above to the Primer?

Also, it would be nice to add links from the other specs mentioned to that summary table.
